e2e/common/world.ts

```diff
@@ -20,10 +53,82 @@ export { P };
 export class World {
   user: User;
   app: AppPage;
+  private token: string = null;
 
   constructor() {
     this.app = new AppPage();
   }
+
+
+  /**
+   * Fetch oauth token from application session store or return cached one.
+   * @returns {Promise<string>}
+   */
+  async getOauthToken(): P<string> {
+    if (this.token == null) {
+      this.token = await this.app.sessionStorage.getItem(OAUTH_TOKEN_NAME);
+    }
+    return this.token;
+  }
+
+
+  async authRequest(urlString: string, method: string = 'GET'): P<IpaasResult> {
+    const link = url.parse(urlString);
+    const deferred: Deferred<IpaasResult> = new Deferred();
+
+    const token = await this.getOauthToken();
+    const options: http.RequestOptions = {
+      method: 'GET',
+      protocol: link.protocol,
+      hostname: link.hostname,
+      path: link.path,
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+    };
+
+    log.info(`invoking http request with options: ${JSON.stringify(options)}`);
+    https.request(options, (res) => {
+      let data = '';
+
+      res.on('data', chunk => {
+        log.debug(`data chunk (${Object.prototype.toString.call(chunk)}: ${chunk}`);
+        data += chunk;
+      });
+
+      res.on('end', () => {
+        log.info(`http request finished with status: ${res.statusCode}`);
+        const result: IpaasResult = {
+          statusCode: res.statusCode,
+          data: data,
+        };
+        deferred.resolve(result);
+      });
+
+    }).end();
+    return deferred.promise;
```

Comment:
I'm biased toward async callbacks, but you may find yourself running into some issues with this. In the UI we are using observables (streams) instead of promises, but promises will be the easiest way to deal with the asynchronous events, and you'll rarely mix the two. For now, if it works and doesn't cause issues then just go with it, we can refactor later if necessary. The priority is in getting everything fully tested.